packages/material-ui-lab/src/MasonryItem/MasonryItem.test.js

```diff
@@ -0,0 +1,133 @@
+            columnSpan: 1,
+            spacing: { xs: 1, sm: 2, md: 3 },
+          },
+          theme,
+        }),
+      ).to.deep.equal({
+        '@media (min-width:0px)': {
+          gridRowEnd: `span ${Math.ceil(100 + Number(theme.spacing(1).replace('px', '')))}`,
+          paddingBottom: Number(theme.spacing(1).replace('px', '')) - 1,
+        },
+        [`@media (min-width:${defaultTheme.breakpoints.values.sm}px)`]: {
+          gridRowEnd: `span ${Math.ceil(100 + Number(theme.spacing(2).replace('px', '')))}`,
+          paddingBottom: Number(theme.spacing(2).replace('px', '')) - 1,
+        },
+        [`@media (min-width:${defaultTheme.breakpoints.values.md}px)`]: {
+          gridRowEnd: `span ${Math.ceil(100 + Number(theme.spacing(3).replace('px', '')))}`,
+          paddingBottom: Number(theme.spacing(3).replace('px', '')) - 1,
+        },
+        width: '100%',
+        [`& > *`]: {
+          width: '100%',
+        },
+        visibility: 'visible',
+        gridColumnEnd: 'span 1',
+      });
+    });
+
+    it('should render with given column span', () => {
+      expect(
+        style({
+          styleProps: {
+            height: 100,
+            columnSpan: 2,
+            spacing: 1,
+          },
+          theme,
+        }),
+      ).to.deep.equal({
+        gridRowEnd: `span ${Math.ceil(100 + Number(theme.spacing(1).replace('px', '')))}`,
+        paddingBottom: Number(theme.spacing(1).replace('px', '')) - 1,
+        width: '100%',
+        [`& > *`]: {
+          width: '100%',
+        },
+        visibility: 'visible',
+        gridColumnEnd: 'span 2',
+      });
+    });
+
+    it('should compute grid-row-end based on given height', () => {
+      const { getByTestId } = render(
+        <MasonryItem height={150} data-testid="test-root">
+          {children}
+        </MasonryItem>,
+      );
+      const computedStyle = getComputedStyle(getByTestId('test-root'));
+      expect(computedStyle['grid-row-end']).to.equal(
+        `span ${Math.ceil(150 + Number(theme.spacing(1).replace('px', '')))}`,
+      );
+    });
+  });
+
+  describe('props:', () => {
+    describe('prop: component', () => {
+      it('should render a div by default', () => {
+        const { container } = render(<MasonryItem>{children}</MasonryItem>);
+        expect(container.firstChild).to.have.property('nodeName', 'DIV');
+      });
+
+      it('should render a different component', () => {
+        const { container } = render(<MasonryItem component="span">{children}</MasonryItem>);
+        expect(container.firstChild).to.have.property('nodeName', 'SPAN');
+      });
+    });
+
+    describe('prop: className', () => {
+      it('should append the className to the root element', () => {
+        const { container } = render(<MasonryItem className="foo">{children}</MasonryItem>);
+        expect(container.firstChild).to.have.class(classes.root);
+        expect(container.firstChild).to.have.class('foo');
+      });
+    });
+  });
```

Comment:
```suggestion
```

Same as in the Masonry component, these are covered in the conformance test suite.